src/main/java/nextstep/subway/domain/Sections.java

```diff
@@ -25,10 +30,152 @@ public Sections(List<Section> sections) {
     }
 
     public void add(Section section) {
+        checkValidation(section);
+        if (isSameUpStation(section)) {
+            updateStationWhenUpStationSame(section);
+        }
+        if (isSameDownStation(section)) {
+            updateStationWhenDownStationSame(section);
+        }
         sections.add(section);
     }
 
+    private boolean hasEndPointIssue(Section newSection) {
+        boolean hasIssue = false;
+        Section firstSection = findFirstSection();
+        Section lastSection = findLastSection();
+        if (firstSection.isEqualUpStationNewSectionDownStation(newSection)
```

Comment:
```suggestion
        return firstSection.isEqualUpStationNewSectionDownStation(newSection) && lastSection.isEqualDownStationNewSectionUpStation(newSection)
```

hasIssue 없이 이렇게만 해주어도 되지 않을까요...?

Reply:
오 넵  boolean 지역 변수도 제거되고 좋네요!  리뷰 감사드립니다.